frontend/appflowy_flutter/lib/user/application/auth/supabase_auth_service.dart

```diff
@@ -229,16 +245,15 @@ Completer<Either<FlowyError, UserProfilePB>> supabaseLoginCompleter({
 
   subscription = auth.onAuthStateChange.listen((event) async {
     final user = event.session?.user;
-    if (event.event != AuthChangeEvent.signedIn || user == null) {
-      completer.complete(left(AuthError.supabaseSignInWithOauthError));
-    } else {
+    if (event.event == AuthChangeEvent.signedIn && user != null) {
       final response = await onSuccess(
         user.id,
         user.email ?? user.newEmail ?? '',
       );
+      // Only cancle the subscription if the Event is signedIn.
```

Comment:
```suggestion
      // Only cancel the subscription if the Event is signedIn.
```

Reply:
Why cancel the subscription only when it's signed in? I think we should cancel it even if the sign-in event fails.

Reply:
The subscription will receive the event upon logout, which can sometimes cause the login to malfunction. Using this function means it only focuses on the login event